### PR TITLE
Lookup usersignup by username first then userid

### PR DESCRIPTION
**Pull request by sbryzak**
This PR makes a small change to lookup the UserSignup by username first instead of userID

Fixes https://issues.redhat.com/browse/CRT-1570

### CODE REVIEW

**Review by MatousJobanek — CHANGES_REQUESTED** [review]
As I mentioned in the comment - we need to fix all broken UserSignups first, then we can change the way we retrieve them.

File: pkg/signup/service/signup_service.go [hunk 1]
```diff
@@ -336,11 +336,11 @@ func (s *ServiceImpl) GetSignup(userID, username string) (*signup.Signup, error)
 // GetUserSignup is used to return the actual UserSignup resource instance, rather than the Signup DTO
 func (s *ServiceImpl) GetUserSignup(userID, username string) (*toolchainv1alpha1.UserSignup, error) {
 	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserIdentifier(userID))
+	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(EncodeUserIdentifier(username))
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/signup/service/signup_service.go:339]: The analysis task has revealed that there are currently no UserSignups stuck in migration, so this is not a blocker.

File: pkg/controller/signup_test.go [hunk 2]
```diff
@@ -576,7 +576,7 @@ func (s *TestSignupSuite) TestVerifyPhoneCodeHandler() {
 
 		require.Equal(s.T(), "Not Found", bodyParams["status"])
 		require.Equal(s.T(), float64(404), bodyParams["code"])
-		require.Equal(s.T(), fmt.Sprintf(" \"%s\" not found: user not found", userSignup.Name), bodyParams["message"])
+		require.Equal(s.T(), " \"\" not found: user not found", bodyParams["message"])
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/controller/signup_test.go:579]: The error is correct, this is what's returned from the Kubernetes client which is passed back to the caller without modification.

[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/controller/signup_test.go:579]: ok, I see why - at line 568 there is missing the username:
https://github.com/codeready-toolchain/registration-service/blob/2924c85484739aca7f778ed628d0271836f6fea4/pkg/controller/signup_test.go#L568
could you please fix that so it uses non-empty username?

**Review by MatousJobanek — APPROVED** [review]
Since there is no usersignup blocked in the migration process, then we should be fine with merging this PR. 
Just please fix the test that I mentioned in the comment https://github.com/codeready-toolchain/registration-service/pull/274#discussion_r935287038

**Pull request merged**